tests/integration/double_vote.go

```diff
@@ -0,0 +1,400 @@
+		valSet,
+		signer,
+		s.consumerChain.ChainID,
+	)
+
+	badVote := makeAndSignVote(
+		blockID2,
+		s.consumerCtx().BlockHeight(),
+		s.consumerCtx().BlockTime(),
+		valSet,
+		signer,
+		s.consumerChain.ChainID,
+	)
+
+	testCases := []struct {
+		name    string
+		ev      *tmtypes.DuplicateVoteEvidence
+		header  *ibctmtypes.Header
+		expPass bool
+	}{
+		{
+			"create infraction header using an invalid consumer chain id - shouldn't pass",
+			&tmtypes.DuplicateVoteEvidence{
+				VoteA:            vote1,
+				VoteB:            badVote,
+				ValidatorPower:   val.VotingPower,
+				TotalVotingPower: val.VotingPower,
+				Timestamp:        s.consumerCtx().BlockTime(),
+			},
+			&ibctmtypes.Header{
+				SignedHeader: &tmproto.SignedHeader{
+					Header: &tmproto.Header{
+						ChainID: "chainID",
+					},
+				},
+			},
+			false,
+		},
+		{
+			// create an invalid evidence containing two identical votes
+			"invalid double voting evidence - shouldn't pass",
+			&tmtypes.DuplicateVoteEvidence{
+				VoteA:            vote1,
+				VoteB:            vote1,
+				ValidatorPower:   val.VotingPower,
+				TotalVotingPower: val.VotingPower,
+				Timestamp:        s.consumerCtx().BlockTime(),
+			},
+			s.consumerChain.LastHeader,
+			false,
+		},
+		{
+			// In order to create an evidence for a consumer chain,
+			// we create two votes that only differs by their Block IDs and
+			// signed them using the same validator and chain ID of the consumer chain
+			"valid double voting evidence - should pass",
+			&tmtypes.DuplicateVoteEvidence{
+				VoteA:            vote1,
+				VoteB:            badVote,
+				ValidatorPower:   val.VotingPower,
+				TotalVotingPower: val.VotingPower,
+				Timestamp:        s.consumerCtx().BlockTime(),
+			},
+			s.consumerChain.LastHeader,
+			true,
+		},
+	}
+
+	consuAddr := types.NewConsumerConsAddress(sdk.ConsAddress(val.Address.Bytes()))
+	provAddr := s.providerApp.GetProviderKeeper().GetProviderAddrFromConsumerAddr(s.providerCtx(), s.consumerChain.ChainID, consuAddr)
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			err = s.providerApp.GetProviderKeeper().HandleConsumerDoubleVoting(
+				s.providerCtx(),
+				tc.ev,
+				tc.header,
+			)
+			if tc.expPass {
+				s.Require().NoError(err)
+
+				// verifies that the jailing and tombstoning has occurred
+				s.Require().True(s.providerApp.GetTestStakingKeeper().IsValidatorJailed(s.providerCtx(), provAddr.ToSdkConsAddr()))
+				s.Require().True(s.providerApp.GetTestSlashingKeeper().IsTombstoned(s.providerCtx(), provAddr.ToSdkConsAddr()))
+			} else {
+				s.Require().Error(err)
+
+				// verifies that no jailing and tombstoning has occurred
+
+				s.Require().False(s.providerApp.GetTestStakingKeeper().IsValidatorJailed(s.providerCtx(), provAddr.ToSdkConsAddr()))
+				s.Require().False(s.providerApp.GetTestSlashingKeeper().IsTombstoned(s.providerCtx(), provAddr.ToSdkConsAddr()))
+			}
+		})
+	}
+}
+
+func (s *CCVTestSuite) TestVerifyDoubleVotingEvidence() {
```

Comment:
I see this more of a UT in double_vote_test.go. Would that be possible or it's too complicated?

Reply:
Changed! Thanks for the input, `TestVerifyDoubleVotingEvidence` runs two times faster now.